src/Logging/Scrubbing/RegexLogScrubbingRule.cs

```diff
@@ -11,25 +11,41 @@ namespace Microsoft.Omex.Extensions.Logging.Scrubbing
 	internal class RegexLogScrubbingRule : ILogScrubbingRule
 	{
 		private readonly Regex m_regexToReplace;
-		private readonly string m_replacementValue;
+		private readonly MatchEvaluator? m_matchEvaluator;
+		private readonly string? m_replacementValue;
 
 		/// <summary>
 		/// Initializes a new instance of the <see cref="RegexLogScrubbingRule"/> class.
 		/// </summary>
 		/// <param name="regexToReplace">The regular expression specifying the strings to replace.</param>
 		/// <param name="replacementValue">The value with which to replace the matching text.</param>
-		public RegexLogScrubbingRule(string regexToReplace, string replacementValue)
+		public RegexLogScrubbingRule(string regexToReplace, string? replacementValue)
```

Comment:
I would argue that if user is colling this constructor `replacementValue` input parameter should not be `null`.  While you still need to keep fields as nullable. Same goes for `matchEvaluator` in another constructor.

Reply:
Yes it was added due to non-nullable leaving constructor. I updated the code there to remove this part.